### PR TITLE
added QR code app, needs eventType.URL

### LINKED ISSUE #4699: QR Code App
add a "Download QR Code" button in event-types

### PR DESCRIPTION
**Pull request by PeerRich**
- [x] add eventType.URL @hariombalhara 

CleanShot 2022-10-17 at 16 49 44@2x

### CODE REVIEW

**Comment by hariombalhara** [comment]
@PeerRich Do you want to pick this up now or I should pick it up ?

Is there a QR code library that you have planned to use?

**Comment by PeerRich** [comment]
> @PeerRich Do you want to pick this up now or I should pick it up ?
> 
> 
> 
> Is there a QR code library that you have planned to use? 

i was thinking of just using one of those public Qr code apis 

will look into it today

File: packages/app-store/qr_code/extensions/EventTypeAppCard.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import type { EventTypeAppCardComponent } from "@calcom/app-store/types";
+import { Tooltip } from "@calcom/ui/v2";
+
+import { appDataSchema } from "../zod";
+
+const EventTypeAppCard: EventTypeAppCardComponent = function EventTypeAppCard({ eventType, app }) {
+  const [getAppData, setAppData] = useAppContextWithSchema<typeof appDataSchema>();
+  const [enabled, setEnabled] = useState(getAppData("enabled"));
+
+  const eventTypeURL = "https://cal.dev/pro/30min"; // TODO: eventType.URL
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply packages/app-store/qr_code/extensions/EventTypeAppCard.tsx]: Sorted

File: packages/app-store/qr_code/zod.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { z } from "zod";
+
+import { eventTypeAppCardZod } from "../eventTypeAppCardZod";
+
+export const appDataSchema = eventTypeAppCardZod.merge(
+  z.object({
+    isSunrise: z.boolean(),
```
Comment (PeerRich) [inline packages/app-store/qr_code/zod.ts:7]: @hariombalhara this can prob. be removed, right?

Comment (hariombalhara) [reply packages/app-store/qr_code/zod.ts]: Oh locally removed. Forgot to push. Done now

**Review by hariombalhara — APPROVED** [review]

**Review by hariombalhara — APPROVED** [review]

File: packages/app-store/qr_code/extensions/EventTypeAppCard.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const [enabled, setEnabled] = useState(getAppData("enabled"));
+
+  const eventTypeURL = eventType.URL;
+
+  function QRCode({ size, data }: { size: number; data: string }) {
+    const QR_URL = "https://api.qrserver.com/v1/create-qr-code/?size=" + size + "&data=" + data;
+    return (
+      <Tooltip content={eventTypeURL}>
+        <a download href={QR_URL} target="_blank" rel="noreferrer">
+          <img
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply packages/app-store/qr_code/extensions/EventTypeAppCard.tsx:21]: not important enough

**Pull request merged**